test/methods.jl

```diff
@@ -329,4 +343,12 @@ end
             init,
         )
     end
+
+    @testset "Missing criterion implementation" begin
+        struct NoCriterion <: RotationMethod{Orthogonal} end
+
+        @test_throws "NoCriterion does not implement" criterion(NoCriterion(), randn(6, 6))
+        # Enzyme.jl would refuse to autodiff because it detects that fallback criterion() throws an error
+        @test_throws "Function to differentiate" criterion_and_gradient!(randn(6, 5), NoCriterion(), randn(6, 5))
```

Comment:
I don't really like that we are relying on Enzyme to catch the error in this edge case. However, I have no idea how to catch them with the current implementation of `criterion` and `criterion_and_gradient!`. Any ideas?

Reply:
We don't rely on Enzyme, it's just too smart and recognizes that the fallback implementation would throw an "not implemented" error, so it throws an error first. :)
So I don't see a problem in terms of the design.
We may need to update this particular test if the Enzyme behavior changes.
We can make it less specific already now, but I don't see it as a big problem.